addons/controllers/addon_controller.go

```diff
@@ -243,6 +252,15 @@ func (r *AddonReconciler) reconcileNormal(
 	log logr.Logger,
 	cluster *clusterapiv1beta1.Cluster) (ctrl.Result, error) {
 
+	_, isManagmentCluster := cluster.Labels[constants.ManagementClusterRoleLabel]
+	if !isManagmentCluster && !controllerutil.ContainsFinalizer(cluster, addontypes.AddonFinalizer) {
```

Comment:
The `!controllerutil.ContainsFinalizer` check seems to be redundant since its there in addFinalizer too.

Reply:
Ack